llrt_core/src/modules/http/response.rs

```diff
@@ -360,25 +388,94 @@ impl<'js> Response<'js> {
         })
     }
 
-    #[qjs(get, rename = "type")]
-    fn response_type(&self) -> &'js str {
-        "basic"
+    #[qjs(static)]
+    fn error(ctx: Ctx<'js>) -> Result<Self> {
+        Ok(Self {
+            body: None,
+            method: "".into(),
+            url: "".into(),
+            start: Instant::now(),
+            status: 0,
+            status_text: None,
+            redirected: false,
+            headers: Class::instance(ctx.clone(), Headers::default())?,
+            body_attributes: BodyAttributes {
+                content_type: None,
+                content_encoding: None,
+            },
+            abort_receiver: None,
+        })
     }
 
-    #[qjs(get)]
-    fn status_text(&self) -> String {
-        if let Some(text) = &self.status_text {
-            return text.to_string();
+    #[qjs(static, rename = "json")]
+    fn json_static(ctx: Ctx<'js>, body: Value<'js>, options: Opt<Object<'js>>) -> Result<Self> {
+        let mut status = 200;
+        let mut headers = None;
+        let mut status_text = None;
+
+        if let Some(opt) = options.0 {
+            if let Some(status_opt) = opt.get("status")? {
+                status = status_opt;
+            }
+            if let Some(headers_opt) = opt.get("headers")? {
+                headers = Some(Headers::from_value(&ctx, headers_opt)?);
+            }
+            if let Some(status_text_opt) = opt.get("statusText")? {
+                status_text = Some(status_text_opt);
+            }
         }
-        STATUS_TEXTS.get(&self.status).unwrap_or(&"").to_string()
+
+        let headers = if let Some(headers) = headers {
```

Comment:
```rust
let headers = Class::instance(ctx.clone(),headers.unwrap_or_default())?;
```

Reply:
Thank you for your review.
We have made the improvements as you just pointed out.